easybuild/framework/easyblock.py

```diff
@@ -3350,6 +3352,271 @@ def _sanity_check_step_multi_deps(self, *args, **kwargs):
+                            # Make sure to use LooseVersion so that e.g. 9.0 < 9.0a < 9.2 < 9.10
+                            highest_cc = [sorted(cfg_ccs, key=LooseVersion)[-1]]
+                            missing_ptx_ccs = list(set(highest_cc) - set(derived_ptx_ccs))
+
+                            if missing_ptx_ccs:
+                                files_missing_ptx.append(path)
+                                num_files_missing_ptx += 1
+                                fail_msg = "Configured highest compute capability was '%s', "
+                                fail_msg += "but no PTX code for this compute capability was found in '%s' "
+                                fail_msg += "(PTX architectures supported in that file: %s). "
+                                if path in ignore_file_list or ignore_failures:
+                                    files_missing_ptx_ignored.append(path)
+                                    num_files_missing_ptx_ignored += 1
+                                    fail_msg += ignore_msg
+                                    self.log.warning(fail_msg, highest_cc[0], path, derived_ptx_ccs)
+                                elif accept_missing_ptx:
+                                    self.log.warning(fail_msg, highest_cc[0], path, derived_ptx_ccs)
+                                else:
+                                    files_missing_ptx_fails.append(path)
+                                    num_files_missing_ptx_fails += 1
+                                    fails.append(fail_msg % (highest_cc[0], path, derived_ptx_ccs))
+                            else:
+                                msg = (f"Output of 'cuobjdump' checked for '{path}'; ptx code was present for (at "
+                                       "least) the highest CUDA compute capability in cuda_compute_capabilities")
+                                self.log.debug(msg)
+                else:
+                    self.log.debug(f"Not sanity checking files in non-existing directory {dirpath}")
+
+            # Summary
+            summary_msg = "CUDA sanity check summary report:\n"
+            summary_msg += f"Number of CUDA files checked: {num_cuda_files}\n"
+            summary_msg += f"Number of files missing one or more CUDA Compute Capabilities: {num_files_missing_cc} "
+            summary_msg += f"(ignored: {num_files_missing_cc_ignored}, fails: {num_files_missing_cc_fails})\n"
+            if accept_ptx_as_cc:
+                summary_msg += "Number of files missing one or more CUDA Compute Capabilities, but has suitable "
+                summary_msg += "PTX code that can be JIT compiled for the requested CUDA Compute Capabilities: "
+                summary_msg += f"{num_files_missing_cc_but_has_ptx}\n"
+            summary_msg += f"Number of files with device code for more CUDA Compute Capabilities than requested: "
+            if strict_cc_check:
```

Comment:
```suggestion
            if strict_cc_check and not ignore_failures:
```

Reply:
I would still print this (like all the other ones): it will simply report that there _are_ files with additional CCs, but that all of those are ignored.

Reply:
Actually: I'll make an extra `elif` with a more specific message that it's not considered a failure because someone is running with `--disable-cuda-sanity-check-error-on-fail`.

Reply:
Fixed, now give a specific message for `if ignore_failures: ... elif strict_cc_check: ... else: ...`